sdk/core/Azure.Core/tests/public/PatchModels/SimplePatchModel.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core.Json;
+
+namespace Azure.Core.Tests.PatchModels
+{
+    /// <summary>
+    /// This model illustrates optional read/write "primitive" properties.
+    /// </summary>
+    public partial class SimplePatchModel
+    {
+#pragma warning disable AZC0020 // Avoid using banned types in libraries
```

Comment:
Only generated patch models should suppress this warning.

Reply:
Can we update the analyzer to allow for this narrow usage?  I would be fine with a follow up tracking item and link the issue here for reference if we don't want to bloat this PR.

Reply:
@m-nash - I'm open to this -- can you describe how we would know that it is being used appropriately in the context of a generated Patch model, and not outside that usage?

Reply:
That I don't know until we finalize on the pattern.  However it could be something along the lines of "only used as input to a PATCH operation"?